Sources/SourceKitLSP/Swift/DiagnosticReportManager.swift

```diff
@@ -86,11 +86,7 @@ actor DiagnosticReportManager {
   }
 
   func removeItemsFromCache(with uri: DocumentURI) async {
-    let tasksToCancel = reportTaskCache.filter { $0.snapshotID.uri == uri }
     reportTaskCache.removeAll(where: { $0.snapshotID.uri == uri })
-    for task in tasksToCancel {
-      await task.reportTask.cancel()
-    }
```

Comment:
I suppose we could still do the cancellation e.g on close (just not re-open), would that be beneficial?